pkg/controllers/management/auth/indexes_test.go

```diff
@@ -0,0 +1,271 @@
+package auth
+
+import (
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_getRBOwnerKey(t *testing.T) {
+	type args struct {
+		rb *v1.RoleBinding
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "get_two_ownerreferences",
+			args: args{
+				rb: &v1.RoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "v1",
+								Kind:       "Pod",
+								Name:       "example-pod",
+								UID:        "12345-67890-abcdef-1",
+							},
+							{
+								APIVersion: "v1",
+								Kind:       "Pod",
+								Name:       "example-pod",
+								UID:        "12345-67890-abcdef-2",
+							},
+						},
+					},
+				},
+			},
+			want: []string{"12345-67890-abcdef-1", "12345-67890-abcdef-2"},
+		},
+		{
+			name: "get_zero_ownerreferences",
+			args: args{
+				rb: &v1.RoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						OwnerReferences: []metav1.OwnerReference{},
+					},
+				},
+			},
+			// slice of 0 length is considered nil.
+			want: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getRBOwnerKey(tt.args.rb); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("getRBOwnerKey() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_rbRoleSubjectKeys(t *testing.T) {
+	type args struct {
+		roleName string
+		subject  []v1.Subject
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "test_with_real_examples",
+			args: args{
+				roleName: "rbac-role-binding-role-binding",
+				subject: []v1.Subject{
+					{
+						Kind: "User",
+						Name: "test1",
+					},
+					{
+						Kind: "ServiceAccount",
+						Name: "test1",
+					},
+				},
+			},
+			want: []string{
+				"rbac-role-binding-role-binding.User.test1",
+				"rbac-role-binding-role-binding.ServiceAccount.test1",
+			},
+		},
+		{
+			name: "test_with_empty_examples",
+			args: args{
+				roleName: "",
+				subject: []v1.Subject{
+					{
+						Kind: "User",
+						Name: "test1",
+					},
+					{
+						Kind: "",
+						Name: "",
+					},
+				},
+			},
+			want: []string{".User.test1", ".."},
+		},
```

Comment:
> I suspect it's never happened, and I can't imagine how it could happen...but... 😄

Yes, I had the same thought, this should never happen. If you agree I could change the function accordingly in order to skip the "empty" (`".."`) cases and return an error instead. wdyt?